### PR TITLE
 Disassembling checks for null recipe - fixes #14780, fixes #14735

### LINKED ISSUE #14735: Attempting to disassemble with no recipe causes a crash
Old `crafting_allowed()` had no arguments, new one has a recipe argument, typically invoked like `crafting_allowed( *recipe_ptr )`. The recipe pointer is null for items that can't be disassembled.

This causes crashes when disassembling items that can't be disassembled.

### LINKED ISSUE #14780: Crash: Trying to Dissasemble a Sewing Kit crashes the game
OS: Windows 10
Build: 0.C-8885-g35ddeeb, upgraded later to latest experimental, still happens
Reproduction Rate: 100%

How to Crash: In inventory, select a sewing kit (charges doesnt matter) and press D to dissasemble, the action is greyed out but it makes you crash

### PR DESCRIPTION
**Pull request by ghost**
Check for lighting was before checking for null pointer which resulted in segfault.

### CODE REVIEW

File: src/crafting.cpp [hunk 1]
```diff
@@ -1302,35 +1308,35 @@ void player::disassemble(item &dis_item, int dis_pos, bool ground)
 
     //checks to see if you're disassembling rotten food, and will stop you if true
     if( (dis_item.is_food() && dis_item.goes_bad()) ||
-        (dis_item.is_food_container() && dis_item.contents[0].goes_bad()) ) {
+            (dis_item.is_food_container() && dis_item.contents[0].goes_bad()) ) {
```
Comment (kevingranade) [inline src/crafting.cpp]: The previous alignment was correct.

**Comment by DanmakuDan** [comment]
Books can't be disassembled now, the function returns early seeing that there is no recipe even though the code handles that case further down.

**Comment by chaosvolt** [comment]
Huh, I missed this. Interesting.

Dare I ask how we got disassemble attempts of items that can't be disassembled? I don't recall that occurring before.

**Comment by pingpong2011** [comment]
I was dissembling a two-by-four for skewers. I thought before it would just say "this item can't be dissembled" if you tried on the wrong object. I might be mixing it up with clothing, because certain cloths definitely used to give warnings that you butcher/cut it instead.

I'm certain there were incredible amount of dissembling things that couldn't be dissembled, simply by the fact you can hit a key you don't intend to. Also, anytime I'd come across some random electronics I didn't want I'd scrap it for parts, and not everything could be crafted before.

**Comment by DanmakuDan** [comment]
There was a rework for mass butcher/disassemble,  #14724.

**Comment by chaosvolt** [comment]
Ah, that explains it.

**Pull request merged**